tfx_addons/example_filter/executor.py

```diff
@@ -0,0 +1,352 @@
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generic TFX model evaluator executor."""
+
+import os
+from typing import Any, Dict, List
+
+from absl import logging
+import apache_beam as beam
+import tensorflow_model_analysis as tfma
+from tensorflow_model_analysis import constants as tfma_constants
+# Need to import the following module so that the fairness indicator post-export
+# metric is registered.
+import tensorflow_model_analysis.addons.fairness.post_export_metrics.fairness_indicators  # pylint: disable=unused-import
+from tfx import types
+from tfx.components.evaluator import constants
+from tfx.components.util import udf_utils
+from tfx.components.util import tfxio_utils
+from tfx.dsl.components.base import base_beam_executor
+from tfx.proto import evaluator_pb2
+from tfx.types import artifact_utils
+from tfx.types import standard_component_specs
+from tfx.utils import io_utils
+from tfx.utils import json_utils
+from tfx.utils import path_utils
+from tfx.utils import proto_utils
+from tfx_bsl.tfxio import tensor_adapter
+
+_TELEMETRY_DESCRIPTORS = ['Evaluator']
+
+
+class Executor(base_beam_executor.BaseBeamExecutor):
```

Comment:
Having a hard time figuring out how is filtering happenning? Seems there's a lot of code here from model evaluator, can we remove extraneous code?

Reply:
I have pushed a commit with the fix